streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -432,13 +501,23 @@ public void shouldPunctuateSystemTimeWhenIntervalElapsed() {
 
     @Test
     public void shouldNotPunctuateSystemTimeWhenIntervalNotElapsed() {
+        task = createStatelessTask(false);
+        task.initialize();
+        assertFalse(task.maybePunctuateSystemTime());
+        processorSystemTime.supplier.checkAndClearPunctuateResult(PunctuationType.WALL_CLOCK_TIME);
+    }
+
+    @Test
+    public void shouldPunctuateOnceSystemTimeAfterGap() {
         task = createStatelessTask(false);
         task.initialize();
         long now = time.milliseconds();
-        assertTrue(task.maybePunctuateSystemTime()); // first time we always punctuate
-        time.sleep(9);
+        time.sleep(100);
+        assertTrue(task.maybePunctuateSystemTime());
         assertFalse(task.maybePunctuateSystemTime());
-        processorSystemTime.supplier.checkAndClearPunctuateResult(PunctuationType.WALL_CLOCK_TIME, now);
+        time.sleep(10);
```

Comment:
we should add one more test, with `sleep(12)` to "pass" the next scheduled punctuation already.